internal/cephfs/controllerserver.go

```diff
@@ -712,3 +712,26 @@ func (cs *ControllerServer) DeleteSnapshot(ctx context.Context, req *csi.DeleteS
 
 	return &csi.DeleteSnapshotResponse{}, nil
 }
+
+// validateSnapshotRetention verifies if the referred subvolume has
+// 'snapshot-retention' feature.
+func validateSnapshotRetention(ctx context.Context, volID volumeID, cr *util.Credentials, volOptions *volumeOptions) (bool, error) {
```

Comment:
make this function more generalized. pass `snapshot-retention` as an argument and check the subvolume feature is present

Reply:
Thanks. Will do it.